### PR TITLE
Az/silhouette improvements

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
## What does this PR do?

- revert bufferview change that would likely lead to invalid row calculations
- avoid per-frame filters and use cached pass pointers instead

## How was this PR tested?

- Tested in custom level in windows editor
- used r_silhouette cvar to turn feature on/off
- turned the camera in game to look away and then back again at the silhouettes to verify there were no issues when out of frame

### CODE REVIEW

**Review by VickyAtAZ — CHANGES_REQUESTED** [review]
I think these pass pointers need to assigned in OnRenderPipelineChanged function so they would be set to nullptr properly in case the render pipeline was removed. 
Also, you may want to set them to nullptr too in the beginning of AddRenderPasses function.

**Comment by carb-lloyd** [comment]
@AMZN-alexpete  - thoughts on the change requests from @VickyAtAZ ?

**Comment by AMZN-alexpete** [comment]
> @AMZN-alexpete - thoughts on the change requests from @VickyAtAZ ?

I think they're good suggestions and I'm testing them now - just waiting on 8k assets to finish processing 😄

**Comment by AMZN-alexpete** [comment]
I've implemented and tested the suggestions on PC in editor - should be good to review.

File: Gems/Atom/Feature/Common/Code/Source/Silhouette/SilhouetteFeatureProcessor.cpp [hunk 1]
```diff
@@ -185,7 +167,35 @@ namespace AZ::Render
...
         {
+            m_compositePass = pass.get();
             renderPipeline->AddPassAfter(pass, postProcessPassName);
         }
     }
+
+    void SilhouetteFeatureProcessor::OnRenderPipelineChanged(AZ::RPI::RenderPipeline* pipeline, [[maybe_unused]] AZ::RPI::SceneNotification::RenderPipelineChangeType changeType)
+    {
+        // update our render pass members in case the pipeline was removed
+        UpdatePasses(pipeline);
```
Comment (VickyAtAZ) [inline Gems/Atom/Feature/Common/Code/Source/Silhouette/SilhouetteFeatureProcessor.cpp:180]: This function is called when a RenderPipeline is removed too. 
you may want to handle that case

**Review by moudgils — APPROVED** [review]

**Review by VickyAtAZ — APPROVED** [review]

**Pull request merged**